app/src/main/java/com/android/shelfLife/ui/authentication/SignIn.kt

```diff
@@ -25,59 +18,61 @@ import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.font.FontWeight
-import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
 import com.android.shelfLife.R
 import com.android.shelfLife.ui.navigation.NavigationActions
-import com.android.shelfLife.ui.navigation.TopLevelDestinations
+import com.android.shelfLife.viewmodel.authentication.SignInState
+import com.android.shelfLife.viewmodel.authentication.SignInViewModel
 import com.google.android.gms.auth.api.signin.GoogleSignIn
 import com.google.android.gms.auth.api.signin.GoogleSignInOptions
-import com.google.android.gms.common.api.ApiException
-import com.google.firebase.Firebase
-import com.google.firebase.auth.AuthResult
-import com.google.firebase.auth.GoogleAuthProvider
-import com.google.firebase.auth.auth
-import com.google.firebase.firestore.FirebaseFirestore
-import com.google.firebase.firestore.SetOptions
-import kotlinx.coroutines.launch
-import kotlinx.coroutines.tasks.await
 
-/**
- * Composable function for the sign-in screen.
- *
- * @param navigationActions The navigation actions to be used in the screen
- */
 @Composable
-fun SignInScreen(navigationActions: NavigationActions) {
+fun SignInScreen(
```

Comment:
There is a problem in this screen . The NavigationActions never get used which means it doesn't lead anywhere